src/test/java/com/tterrag/registrate/test/mod/TestMod.java

```diff
@@ -372,41 +356,54 @@ public ResourceLocation getFlowingTexture() {
 //    private final BlockBuilder<Block, Registrate> INVALID_TEST = registrate.object("invalid")
 //            .block(Block::new)
 //            .addLayer(() -> RenderType::getTranslucent);
-    
+
     private static <T extends Block, P> @NonnullType BlockBuilder<T, P> applyDiamondDrop(BlockBuilder<T, P> builder) {
         return builder.loot((prov, block) -> prov.dropOther(block, Items.DIAMOND));
     }
 
     public TestMod() {
-        
+
         registrate.addRawLang("testmod.custom.lang", "Test");
         registrate.addLang("tooltip", testblock.getId(), "Egg.");
         registrate.addLang("item", testitem.getId(), "testextra", "Magic!");
         registrate.addDataGenerator(ProviderType.ADVANCEMENT, adv -> {
             Advancement.Builder.advancement()
                 .addCriterion("has_egg", InventoryChangeTrigger.TriggerInstance.hasItems(Items.EGG))
                 .display(Items.EGG,
-                        adv.title(registrate.getModid(), "root", "Test Advancement"), adv.desc(registrate.getModid(), "root", "Get an egg."), 
+                        adv.title(registrate.getModid(), "root", "Test Advancement"), adv.desc(registrate.getModid(), "root", "Get an egg."),
                         new ResourceLocation("textures/gui/advancements/backgrounds/stone.png"), FrameType.TASK, true, true, false)
                 .save(adv, registrate.getModid() + ":root");
         });
-        
+
         FMLJavaModLoadingContext.get().getModEventBus().addListener(this::onCommonSetup);
+        FMLJavaModLoadingContext.get().getModEventBus().addListener(this::registerCreativeTab);
         MinecraftForge.EVENT_BUS.addListener(this::afterServerStart);
     }
-    
+
     private void onCommonSetup(FMLCommonSetupEvent event) {
         if (!sawCallback.get()) {
             throw new IllegalStateException("Register callback not fired!");
         }
-        
+
         testblock.asStack();
         testitem.is(Items.SNOWBALL);
         testblockitem.is(Items.STONE);
         testblockbe.is(BlockEntityType.CHEST);
         // testbiome.is(Feature.BAMBOO); // should not compile
     }
-    
+
     private void afterServerStart(ServerStartedEvent event) {
     }
+
+    private void registerCreativeTab(CreativeModeTabEvent.Register event) {
```

Comment:
Hm, I'd prefer something more clever here. I like the ability to set the tab when building the item, perhaps Registrate should accept a registered tab instance or a RL and manage lists internally that it exposes to the tab implementation. Plus running a mapped stream for every call to displayItems seems inefficient.

Reply:
Been unsure on how to implement this, so I have pushed what I have currently implemented to a new different branch, rather than merging it into this PR as I'm unsure if its best solution or way about allowing users to create, modify & register CreativeModeTabs

Link to commit: https://github.com/ApexModder/Registrate/commit/8012896322543a5493a7544d378ec23a63522f50

Reply:
The old legacy methods used to assign CreativeModeTabs to Items have been implemented and marked as deprecated, they function exactly as they did in 1.19.2.

using `#creativeModeTab` in `AbstractRegistrate` marks a tab as default and is assigned to all ItemBuilders by default
using `#tab` in `ItemBuilder` overrides the currently assigned tab.

Items are added to the currently assigned tab via the `CreativeModeTabEvent.BuildContents` event, registered in `ItemBuilder#onBuildCreativeModeTabContents`